frame/contracts/src/exec.rs

```diff
@@ -686,8 +686,11 @@ where
 					contract
 				} else {
 					<ContractInfoOf<T>>::get(&dest)
-						.and_then(|contract| contract.get_alive())
-						.ok_or((Error::<T>::NotCallable.into(), 0))?
+						.ok_or((<Error<T>>::ContractNotFound.into(), 0))
+						.and_then(|contract|
+							contract.get_alive()
+								.ok_or((<Error<T>>::ContractIsTombstone.into(), 0))
+						)?
```

Comment:
```suggestion
						.and_then(ContractInfo::get_alive)
						.ok_or((<Error<T>>::ContractIsTombstone.into(), 0))?
```

Reply:
This won't work because you would return an `Option` from `and_then` but it expects a `Result`.

Reply:
My bad, I got myself confused between `Option` and `Result` :D